core/jobs/decorators/validation_decorators.py

```diff
@@ -210,19 +259,26 @@ def get_id_referencing_properties_by_kind_of_possessor(cls):
             (ModelProperty, set(kind of models)), grouped by the kind of model
             the properties belong to.
         """
-        by_kind = lambda model_property: model_property.model_kind
+        by_kind: (
+            Callable[[model_property.ModelProperty], str]
+        ) = lambda model_property: model_property.model_kind
         id_referencing_properties_by_kind_of_possessor = itertools.groupby(
             sorted(cls._ID_REFERENCING_PROPERTIES.keys(), key=by_kind),
             key=by_kind)
-        references_of = lambda p: cls._ID_REFERENCING_PROPERTIES[p]
-        return {
-            kind: tuple((p, tuple(references_of(p))) for p in properties)
-            for kind, properties in (
-                id_referencing_properties_by_kind_of_possessor)
-        }
+        references_of: (
+            Callable[[model_property.ModelProperty], Set[str]]
+        ) = lambda p: cls._ID_REFERENCING_PROPERTIES[p]
+        return cast(
+            ModelPropertyAndModelKindType,
+            {
+                kind: tuple((p, tuple(references_of(p))) for p in properties)
+                    for kind, properties in (
+                        id_referencing_properties_by_kind_of_possessor)
+            }
+        )
```

Comment:
The explanation for why I used cast here:

Considering the above case, The return type of `references_of(p)` is `Set[str]` and the return type of `tuple(references_of(p))` is Tuple[str, ...] but it should be `Tuple[str]`. Similar goes for `tuple((p, tuple(references_of(p)))` becasue the expected return type is `Dict[str, Tuple[Tuple[ModelProperty, Tuple[str]]]]` but mypy assumes its type as `Dict[str, Tuple[Tuple[ModelProperty, Tuple[str, ...]], ...]]`

Reply:
ohh really sorry, actually I misinterpreted the concept of Tuple[str, ...].
I was assuming that the `Tuple[str, ...]` is a tuple of string value and any other type ( because of `...` in the type ). But in actuality `Tuple[str, ...]` points to the tuple where each value is of type string. That's why I'm assuming that the return type is wrong `Dict[str, Tuple[Tuple[ModelProperty, Tuple[str, ...]], ...]]` according to the docstring.

But now the concept is clear and cast is removed!!